### PR TITLE
fix: Model Registration with BYO scripts

### PR DESCRIPTION
**Pull request by sreedes**
*Issue #, if available:*

*Description of changes:*
This change would enable Registration of Models with BYO script onto the Model Registry. It sets the appropriate environment variables, so that when the model is deployed in an endpoint or used in a transformer it works well.

*Testing done:*
Unit tests and Integ test successful.
## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [X] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [X] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [X] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [X] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/model.py [hunk 1]
```diff
@@ -178,21 +178,23 @@ def register(
...
+        container_def = self.prepare_container_def()
 
         model_pkg_args = sagemaker.get_model_package_args(
             content_types,
             response_types,
             inference_instances,
             transform_instances,
-            model_package_name,
-            model_package_group_name,
-            self.model_data,
```
Comment (mufaddal-rohawala) [inline src/sagemaker/model.py:189]: Is it intentional to remove this argument? I think we need this.

**Comment by mufaddal-rohawala** [comment]
@sreedes is it possible to add tests for validation checks if Model registration now works with BYO Scripts?

File: src/sagemaker/model.py (hunk 1 above)
Comment (sreedes) [reply src/sagemaker/model.py:189]: Actually they are not removed, but passed in as named parameters. This is required as some of the parameters are optional. Also the model_data and image_url are now packed within container_def.

**Comment by sreedes** [comment]
> @sreedes is it possible to add tests for validation checks if Model registration now works with BYO Scripts?

Done.

**Comment by sreedes** [comment]
> @sreedes is it possible to add tests for validation checks if Model registration now works with BYO Scripts?

Done.

File: src/sagemaker/model.py (hunk 1 above)
Comment (mufaddal-rohawala) [reply src/sagemaker/model.py:189]: ok thanks, didn't know `container_def` makes use of model_data and image_uri.

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by jeniyat — APPROVED** [review]

**Pull request merged**